pkg/controller/certificaterequests/selfsigned/selfsigned_test.go

```diff
@@ -0,0 +1,347 @@
+				},
+			},
+			Err: false,
+			CheckFn: func(t *testing.T, s *selfsignedFixture, args ...interface{}) {
+			},
+		},
+		"a valid RSA key should sign a self signed certificate": {
+			Issuer: gen.Issuer("selfsigned-issuer",
+				gen.SetIssuerSelfSigned(v1alpha1.SelfSignedIssuer{}),
+			),
+			CertificateRequest: gen.CertificateRequest("test-cr",
+				gen.AddCertificateRequestAnnotations(map[string]string{
+					v1alpha1.CRPrivateKeyAnnotationKey: "test-rsa-key",
+				}),
+				gen.SetCertificateRequestCSR(csrBytes),
+			),
+			Builder: &testpkg.Builder{
+				KubeObjects:        []runtime.Object{rsaKeySecret},
+				CertManagerObjects: []runtime.Object{},
+			},
+			Err: false,
+			CheckFn: func(t *testing.T, s *selfsignedFixture, args ...interface{}) {
+				resp := args[1].(*issuer.IssueResponse)
+
+				// CA and cert should be the same.
+				if !bytes.Equal(resp.CA, resp.Certificate) {
+					t.Errorf("expected CA and cert to be the same but got:\nCA: %s\nCert: %s",
+						resp.CA, resp.Certificate)
+				}
+
+				// No private key should be returned.
+				if len(resp.PrivateKey) > 0 {
+					t.Errorf("expected to private key returned but got: %s",
+						resp.PrivateKey)
+				}
+			},
+		},
+		"a valid ECDSA key should sign a self signed certificate": {
+			Issuer: gen.Issuer("selfsigned-issuer",
+				gen.SetIssuerSelfSigned(v1alpha1.SelfSignedIssuer{}),
+			),
+			CertificateRequest: gen.CertificateRequest("test-cr",
+				gen.AddCertificateRequestAnnotations(map[string]string{
+					v1alpha1.CRPrivateKeyAnnotationKey: "test-ecdsa-key",
+				}),
+				gen.SetCertificateRequestCSR(csrECBytes),
+			),
+			Builder: &testpkg.Builder{
+				KubeObjects:        []runtime.Object{ecKeySecret},
+				CertManagerObjects: []runtime.Object{},
+			},
+			Err: false,
+			CheckFn: func(t *testing.T, s *selfsignedFixture, args ...interface{}) {
+				resp := args[1].(*issuer.IssueResponse)
+				if resp == nil {
+					t.Errorf("expected a response but got: %+v",
+						args[1])
+					return
+				}
+
+				// CA and cert should be the same.
+				if !bytes.Equal(resp.CA, resp.Certificate) {
+					t.Errorf("expected CA and cert to be the same but got:\nCA: %s\nCert: %s",
+						resp.CA, resp.Certificate)
+				}
+
+				// No private key should be returned.
+				if len(resp.PrivateKey) > 0 {
+					t.Errorf("expected to private key returned but got: %s",
+						resp.PrivateKey)
+				}
+			},
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			if test.Builder == nil {
+				test.Builder = &testpkg.Builder{}
+			}
+
+			test.Setup(t)
+			crCopy := test.CertificateRequest.DeepCopy()
+			resp, err := test.SelfSigned.Sign(test.Ctx, crCopy)
+			if err != nil && !test.Err {
+				t.Errorf("Expected function to not error, but got: %v", err)
+			}
+			if err == nil && test.Err {
+				t.Errorf("Expected function to get an error, but got: %v", err)
+			}
+
+			test.Finish(t, crCopy, resp, err)
```

Comment:
Mind updating these tests to not require a `selfSignedFixture` structure (similar to the CA implementation now)